bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/wrappers/veneer/BigtableHBaseVeneerSettings.java

```diff
@@ -759,59 +736,62 @@ private void configureRetryableCallSettings(
 
   private void configureNonRetryableCallSettings(
       UnaryCallSettings.Builder<?, ?> unaryCallSettings, OperationTimeouts operationTimeouts) {
-    unaryCallSettings.setRetryableCodes(Collections.<StatusCode.Code>emptySet());
+    unaryCallSettings.setRetryableCodes(Collections.emptySet());
 
-    // NOTE: attempt timeouts are not configured for non-retriable rpcs
     if (operationTimeouts.getOperationTimeout().isPresent()) {
       unaryCallSettings
           .retrySettings()
           .setLogicalTimeout(operationTimeouts.getOperationTimeout().get());
     }
   }
 
+  /**
+   * Build a source of truth for timeout related settings.
+   *
+   * <p>This will combine defaults with end user provided overrides to configure response, attempt
+   * and operation timeouts. There 3 types of operations that are configured here:
+   *
+   * <ul>
+   *   <li>Unary - ReadRow, MutateRow, CheckAndMutate, ReadModifyWrite
+   *   <li>Scans - ReadRows (and BulkReadRow)
+   *   <li>Bulk Mutations
+   * </ul>
+   */
   private ClientOperationTimeouts buildCallSettings() {
-    boolean useTimeouts = configuration.getBoolean(BIGTABLE_USE_TIMEOUTS_KEY, true);
-
-    Optional<Duration> DEFAULT_BATCH_BULK_MUTATE_OVERALL_TIMEOUT =
-        Optional.of(Duration.ofMinutes(20));
-    // Set 20 minutes default timeout for batch jobs.
-    // The default override the deprecated BIGTABLE_LONG_RPC_TIMEOUT_MS_KEY.
-    Optional<Duration> bulkMutateOverallTimeout =
-        configuration.getBoolean(BIGTABLE_USE_BATCH, false)
-            ? extractOverallTimeout(BIGTABLE_MUTATE_RPC_TIMEOUT_MS_KEY)
-                .or(DEFAULT_BATCH_BULK_MUTATE_OVERALL_TIMEOUT)
-            : extractOverallTimeout(
-                BIGTABLE_MUTATE_RPC_TIMEOUT_MS_KEY,
-                BigtableOptionsFactory.BIGTABLE_LONG_RPC_TIMEOUT_MS_KEY);
+    Optional<Duration> defaultBatchMutateOverallTimeout =
+        DEFAULT_TIMEOUTS.bulkMutateTimeouts.operationTimeout;
+    if (configuration.getBoolean(BIGTABLE_USE_BATCH, false)) {
+      defaultBatchMutateOverallTimeout = Optional.of(Duration.ofMinutes(20));
+    }
     OperationTimeouts bulkMutateTimeouts =
         new OperationTimeouts(
-            Optional.<Duration>absent(),
-            extractUnaryAttemptTimeout(BIGTABLE_MUTATE_RPC_ATTEMPT_TIMEOUT_MS_KEY),
-            bulkMutateOverallTimeout);
+            DEFAULT_TIMEOUTS.bulkMutateTimeouts.responseTimeout,
+            extractDuration(BIGTABLE_MUTATE_RPC_ATTEMPT_TIMEOUT_MS_KEY)
+                .or(DEFAULT_TIMEOUTS.bulkMutateTimeouts.attemptTimeout),
+            extractOverallTimeout(
+                    BIGTABLE_MUTATE_RPC_TIMEOUT_MS_KEY,
+                    BigtableOptionsFactory.BIGTABLE_LONG_RPC_TIMEOUT_MS_KEY)
```

Comment:
I wonder if we should start ignoring BIGTABLE_LONG_RPC_TIMEOUT_MS_KEY (and the other ones) as well. I think it's already marked as `@drepcated` in the BigtableOptionsFactory, and I find it a bit confusing.. 

But I'm fine with leaving it for now just so that we can get this done :)

Reply:
Lets this as is for now. I think we will need to take a pass to clean up all of the settings. For now, this PR just tries make the existing settings with veneer